### ISSUE TITLE
Improve accessibility on `MentionSuggestions` component

### ISSUE THREAD

**Issue by dugancathal** [draft-js-plugins/draft-js-plugins#2900]
## Describe the feature

I would love to see the Mention plugin provide more context for users interacting with editors via VoiceOver or other, non-visual means.

## Motivation

Improving accessibility of editor plugins for all users.

## Possible implementations

By providing some more ARIA hints on the components in the plugin, we can provide an experience that tells the user more about where they are and how they can interact with it. For example, one inspiration might be how Slack's mentions read:

https://user-images.githubusercontent.com/1172512/183518096-97379059-a90b-4f45-a2e8-2eaae9b9eb70.mov

## Related Issues

I have a POC available with the `@draft-js-plugins/mention` package [here](https://github.com/dugancathal/draft-js-plugins/tree/improve-a11y-on-mention-suggestions)

**Comment by fxOne** [comment]
Nice, can you create a pull request?

**Comment by dugancathal** [comment]
Certainly! Done: https://github.com/draft-js-plugins/draft-js-plugins/pull/2933
